bundles/org.openhab.binding.octopusenergy/src/main/java/org/openhab/binding/octopusenergy/internal/handler/OctopusEnergyBridgeHandler.java

```diff
@@ -0,0 +1,225 @@
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.octopusenergy.internal.OctopusEnergyApiHelper;
+import org.openhab.binding.octopusenergy.internal.OctopusEnergyConfiguration;
+import org.openhab.binding.octopusenergy.internal.OctopusEnergyDiscoveryService;
+import org.openhab.binding.octopusenergy.internal.dto.ElectricityMeterPoint;
+import org.openhab.binding.octopusenergy.internal.dto.GasMeterPoint;
+import org.openhab.binding.octopusenergy.internal.exception.ApiException;
+import org.openhab.binding.octopusenergy.internal.exception.AuthenticationException;
+import org.openhab.binding.octopusenergy.internal.exception.RecordNotFoundException;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OctopusEnergyBridgeHandler} is responsible for handling the bridge things created to use the Octopus
+ * Energy API.
+ *
+ * It also spawns a background polling thread to update the data at a specified time interval.
+ *
+ * @author Rene Scherer - Initial Contribution
+ */
+@NonNullByDefault
+public class OctopusEnergyBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OctopusEnergyBridgeHandler.class);
+
+    private final OctopusEnergyApiHelper apiHelper;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private ZonedDateTime lastRefreshTime = UNDEFINED_TIME;
+
+    public OctopusEnergyBridgeHandler(Bridge bridge, OctopusEnergyApiHelper apiHelper) {
+        super(bridge);
+        this.apiHelper = apiHelper;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Octopus Energy bridge handler.");
+        OctopusEnergyConfiguration config = getConfigAs(OctopusEnergyConfiguration.class);
+
+        if (config.accountNumber.equals("") || config.apiKey.equals("")) {
+            logger.warn("Setting thing '{}' to OFFLINE: Parameter 'accountNumber' and 'apiKey' must be configured.",
+                    getThing().getUID());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-missing-account-number-or-api-key");
```

Comment:
Dont use the logger when changing the thing status, it uses more resources to log twice as all thingstatus' go into the event.log

Please search and change all cases of this, thanks.

Reply:
done